rapid7/ReflectiveDLLInjection
# Fix rapid7/metasploit-framework#16493, remove RWX section and apply section Characteristics

This is a step in the right direction for fixing https://github.com/rapid7/metasploit-framework/issues/16493
This removes *one* of the two RWX regions created when the ReflectiveLoader loads a dll from memory.
The other one (I think is this: https://github.com/rapid7/metasploit-framework/blob/f737e78b876711f1a8d05809fd4632bc95181e65/lib/msf/core/payload/windows/reflective_pe_loader.rb#L32) we can remove later (I suspect a small section needs to be RX, and the rest can be R or RW), additionally we may be able to deallocate this entire block afterwards.

## Verification steps

1) Build meterpreter with these changes
2) Get a meterpreter session (I tested windows/x64/meterpreter/reverse_tcp on Windows 10).
3) View the meterpreter process with process hacker, and look at the Memory tab. Notice one of the RWX allocation is now RW, with one section allocated as RX

## Before

![VirtualBox_Windows 10_25_04_2022_04_28_01](https://user-images.githubusercontent.com/684924/165045354-66601fa5-2ffd-49a5-95e2-d40497659b11.png)

## After
![VirtualBox_Windows 10_25_04_2022_08_57_24](https://user-images.githubusercontent.com/684924/165045390-b5fbb582-a499-4957-a71e-a233fef08c14.png)
